File: app/src/main/java/net/ginteam/carmen/presenter/auth/SignInPresenter.java
package net.ginteam.carmen.presenter.auth;

import android.content.Intent;

import net.ginteam.carmen.contract.auth.SignInContract;
import net.ginteam.carmen.model.auth.UserModel;
import net.ginteam.carmen.provider.ModelCallback;
import net.ginteam.carmen.provider.auth.AuthProvider;

/**
 * Created by Eugene on 1/4/17.
 */

public class SignInPresenter implements SignInContract.Presenter {

    private SignInContract.View mView;

    @Override
    public void signIn(String email, String password) {
        mView.showLoading(true);

        AuthProvider
                .getInstance()
                .userLogin(email, password, new ModelCallback<UserModel>() {
                    @Override
                    public void onSuccess(UserModel resultModel) {
                        mView.showLoading(false);
                        mView.showMain();
                    }

                    @Override
                    public void onFailure(String message) {
                        mView.showLoading(false);
                        mView.showError(message);
                    }
                });
    }

    @Override
    public void requestGoogleAuthentication() {

    }

    @Override
    public void requestFacebookAuthentication() {

    }

    @Override
    public void onActivityResult(int requestCode, int resultCode, Intent data) {

    }

    @Override
    public void attachView(SignInContract.View view) {
        mView = view;
    }

    @Override
    public void detachView() {
        mView = null;
    }

}

File: app/src/main/kotlin/net/ginteam/carmen/kotlin/presenter/authentication/SignInPresenter.kt
package net.ginteam.carmen.kotlin.presenter.authentication

import net.ginteam.carmen.R
import net.ginteam.carmen.kotlin.api.response.ModelSubscriber
import net.ginteam.carmen.kotlin.contract.SignInContract
import net.ginteam.carmen.kotlin.manager.SharedPreferencesManager
import net.ginteam.carmen.kotlin.model.AuthModel
import net.ginteam.carmen.kotlin.presenter.BasePresenter
import net.ginteam.carmen.kotlin.provider.AuthProvider
import net.ginteam.carmen.kotlin.provider.AuthenticationProvider

/**
 * Created by eugene_shcherbinock on 2/15/17.
 */
class SignInPresenter : BasePresenter <SignInContract.View>(), SignInContract.Presenter {

    private val mAuthProvider: AuthProvider = AuthenticationProvider

    override fun signIn(email: String, password: String) {
        mView?.showLoading(true, R.string.authorization_progress_message)

        mAuthProvider
                .userLogin(email, password)
                .subscribe(object : ModelSubscriber <AuthModel>() {
                    override fun success(model: AuthModel) {
                        mView?.showLoading(false)
                        mView?.showMainActivity()
                    }

                    override fun error(message: String, isNetworkError: Boolean) {
                        mView?.showError(message, isNetworkError)
                    }
                })
    }

    override fun isUserSignedInSuccessfully(): Boolean
            = SharedPreferencesManager.userAccessToken.isNotEmpty()

    override fun facebookSignIn() {
        throw UnsupportedOperationException("not implemented")
    }

    override fun googleSignIn() {
        throw UnsupportedOperationException("not implemented")
    }

}